sessionctx/variable/tidb_vars.go

```diff
@@ -132,7 +138,10 @@ const (
 	DefCurretTS                   = 0
 	DefMaxChunkSize               = 1024
 	DefDMLBatchSize               = 20000
-	DefMemThreshold               = 32 * 1024 * 1024 * 1024 // 32 GiB
+	DefTIDBMemQuotaQuery          = 32 * 1024 * 1024 * 1024 // 32GB.
```

Comment:
Can we get the memory of the machine while tidb-server starting and use 80% * memory-size as  the default threshold?

Reply:
https://github.com/shirou/gopsutil this library can satisfy our requirements, But I'm afraid that some architectures are not supported.